contracts/HubPool.sol

```diff
@@ -231,6 +248,33 @@ contract HubPool is HubPoolInterface, Testable, Lockable, MultiCaller, Ownable {
      *                ADMIN FUNCTIONS                *
      *************************************************/
 
+    /**
+     * @notice Pauses the bundle proposal and execution process. This is intended to be used during upgrades or when
+     * something goes awry.
+     * @param pause true if the call is meant to pause the system, false if the call is meant to unpause it.
+     */
+    function setPaused(bool pause) public onlyOwner nonReentrant {
+        paused = pause;
+        emit Paused(pause);
+    }
+
+    /**
+     * @notice This allows for the deletion of the active proposal in case of emergency.
+     * @dev This is primarily intended to rectify situations where an unexecutable bundle gets through liveness in the
+     * case of a non-malicious bug in the proposal/dispute code. Without this function, the contract would be
+     * indefinitely blocked, migration would be required, and in-progress transfers would never be repaid.
+     */
+    function emergencyDeleteProposal() public onlyOwner nonReentrant {
```

Comment:
What if the `poolRebalanceLeafCount == 0` and a new proposal can come in? (i.e. the situation where `emeregencyDeleteProposal()` is called but `executeProposal` has already happened. Should we check for that or is this not harmful?

Reply:
I think no because this is called by the owner. it's up to the owner to ensure this won't break anything.

Reply:
This saves a require statement but is maybe worth a comment explaining why we don't do any checks

Reply:
I don't think this really breaks anything. If poolRebalanceLeafCount == 0, the proposal could be deleted by just making a new proposal. So I don't really think adding that check here protects against anything.